components/clarity-repl/src/repl/docs.rs

```diff
@@ -0,0 +1,70 @@
+use std::collections::HashMap;
+use std::sync::LazyLock;
+
+use clarity::vm::docs::{
+    make_api_reference, make_define_reference, make_keyword_reference, FunctionAPI,
+};
+use clarity::vm::functions::define::DefineFunctions;
+use clarity::vm::functions::NativeFunctions;
+use clarity::vm::variables::NativeVariables;
+
+fn normalize_description(description: &str) -> String {
+    description.replace('\n', " ")
+}
+
+fn format_api_doc(function_api: &FunctionAPI) -> (String, String) {
+    let description = normalize_description(&function_api.description);
+    let doc = format!(
+        "Usage\n{}\n\nDescription\n{}\n\nExamples\n{}",
+        function_api.signature, description, function_api.example
+    );
+    (function_api.name.clone(), doc)
+}
+
+fn build_api_reference() -> HashMap<String, String> {
+    NativeFunctions::ALL
+        .iter()
+        .map(|func| format_api_doc(&make_api_reference(func)))
+        .chain(
+            DefineFunctions::ALL
+                .iter()
+                .map(|func| format_api_doc(&make_define_reference(func))),
+        )
+        .collect()
+}
+
+fn clarity_keywords() -> HashMap<String, String> {
+    NativeVariables::ALL
+        .iter()
+        .filter_map(|func| {
+            make_keyword_reference(func).map(|key| {
+                let description = normalize_description(key.description);
+                let doc = format!("Description\n{}\n\nExamples\n{}", description, key.example);
```

Comment:
Clippy is gonna make you inline this. Also I think the `indoc` macros are preferable for multiline strings:

```suggestion
                let doc = formatdoc!("
                    Description
                    {description}
                    
                    Examples
                    {}", key.example);
```

Reply:
@jbencin Clippy does run the check in the CI.
It's not complaining as long as one of the argument can not be inlined in the brackets (`key.example`)